src/github/entities/user-installation.entity.ts

```diff
@@ -0,0 +1,15 @@
+import { Base } from 'src/universal/base.entity';
+import { Column, Entity, ManyToOne } from 'typeorm';
+import { User } from './user.entity';
+
+@Entity()
+export class UserInstallation extends Base {
+  @Column({ type: 'varchar', length: 20 })
```

Comment:
PullProofAI

The `UserInstallation` entity is missing validation for the `installationId` and `organizationName` fields, which can lead to potential data integrity issues if invalid or unexpected values are stored.

```diff
+ import { IsNotEmpty, IsString, MaxLength } from 'class-validator';

  @Column({ type: 'varchar', length: 20 })
  installationId: string;


  @Column({ type: 'varchar', length: 255 })
  organizationName: string;
```

Adding validation decorators like `IsNotEmpty`, `IsString`, and `MaxLength` will ensure that only valid data is stored in these fields, improving data quality and preventing unexpected behavior.

This change is necessary to enforce data integrity and prevent potential issues arising from storing incorrect or incomplete data in the database. It helps maintain consistency and reliability in the application's data handling.


<!-- This is an auto-generated comment by PullProofAI -->